src/main/kotlin/no/nav/klage/oppgave/egenansatt/EgenAnsattHealthIndicator.kt

```diff
@@ -0,0 +1,33 @@
+package no.nav.klage.oppgave.egenansatt
+
+import no.nav.klage.oppgave.util.getLogger
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.context.event.EventListener
+import org.springframework.kafka.event.ListenerContainerIdleEvent
+import org.springframework.stereotype.Component
+
+@Component
+class EgenAnsattHealthIndicator : HealthIndicator {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+    }
+
+    private var kafkaConsumerHasReadAllMsgs = false;
+
+
+    @EventListener
+    fun eventHandler(event: ListenerContainerIdleEvent) {
+        logger.debug("Received ListenerContainerIdleEvent event")
+        kafkaConsumerHasReadAllMsgs = true
```

Comment:
Jeg skjønte ikke helt denne. Er det en state man ender opp når innlesning gått bra?

Reply:
Ja. Det kommer en ListenerContainerIdleEvent når Kafka-listeneren står idle i en gitt periode. Mao ikke får nye meldinger. Det betyr jo da at man har spolt gjennom alle meldingene som var der. Så har jeg laget det som en en-veis switch, så den skrur seg aldri av igjen. Den er bare ment for oppstart.

Reply:
Jeg var egentlig ganske fornøyd med denne! :-D